selfdrive/loggerd/loggerd.cc

```diff
@@ -298,89 +217,92 @@ void encoder_thread(RotateState *rotate_state, bool is_streaming, bool raw_clips
         break;
       }
 
-      //uint64_t current_time = nanos_since_boot();
-      //uint64_t diff = current_time - extra.timestamp_eof;
-      //double msdiff = (double) diff / 1000000.0;
-      // printf("logger latency to tsEof: %f\n", msdiff);
-
-      { // all the rotation stuff
-
-        pthread_mutex_lock(&s.rotate_lock);
-        pthread_mutex_unlock(&s.rotate_lock);
-
-        // wait if camera pkt id is older than stream
-        rotate_state->waitLogThread();
-
-        if (do_exit) break;
+      // Rules of rotation:
+      // 1.rotate all cameras on first time run.
+      // 2.rotate all cameras after main thread's time-based rotation occur.
```

Comment:
I am a bit confused about why No.2 is needed since we already have No.4?

Reply:
Yes, I did not write the comment clearly